code/modules/lighting/lighting_corner.dm

```diff
@@ -0,0 +1,131 @@
+/var/list/datum/lighting_corner/all_lighting_corners = list()
+/var/datum/lighting_corner/dummy/dummy_lighting_corner = new
+// Because we can control each corner of every lighting overlay.
+// And corners get shared between multiple turfs (unless you're on the corners of the map, then 1 corner doesn't).
+// For the record: these should never ever ever be deleted, even if the turf doesn't have dynamic lighting.
+
+// This list is what the code that assigns corners listens to, the order in this list is the order in which corners are added to the /turf/corners list.
+/var/list/LIGHTING_CORNER_DIAGONAL = list(NORTHEAST, SOUTHEAST, SOUTHWEST, NORTHWEST)
+
+/datum/lighting_corner
+	var/list/turf/masters                 = list()
+	var/list/datum/light_source/affecting = list() // Light sources affecting us.
+	var/active                            = FALSE  // TRUE if one of our masters has dynamic lighting.
+
+	var/x     = 0
+	var/y     = 0
+	var/z     = 0
+
+	var/lum_r = 0
+	var/lum_g = 0
+	var/lum_b = 0
+
+	var/needs_update = FALSE
+
+	var/cache_r  = LIGHTING_SOFT_THRESHOLD
+	var/cache_g  = LIGHTING_SOFT_THRESHOLD
+	var/cache_b  = LIGHTING_SOFT_THRESHOLD
+	var/cache_mx = 0
+
+	var/update_gen = 0
+
+/datum/lighting_corner/New(var/turf/new_turf, var/diagonal)
```

Comment:
@Fox-McCloud I'm not sure it's a good idea to change these- this file as it is is nearly 100% whole-copied from /vg/